### ISSUE TITLE
Accessibility improvements to Piwik

### ISSUE THREAD

**Issue by mattab** [piwik/piwik#4167]
We spent really interesting time with Julius, an accessibility specialist who is blind and agreed to test the piwik.org website and the software. 

**How it works**

Using a special Screen Reading (Text Reading) software that lets him browse the page, he can go from heading to heading (and optionally go within to read text), go through menus and nested lists, each time the voice will read the label of the element (and read 'link' if the label is clickable), including hidden elements (that would normally display on hover etc).

The screen reader will also read the title of the window, let him look at the list of all links within the page and quickly access one, among many features (all accessible via keyboard shortcuts). 

**Improving the accessibility of Piwik platform**
- turn the following fake buttons into real links: Calendar + All visits segment + Widget & dashboard
- let's link to relevant user documentation from within the software report pages (or within the report inline doc)
- Column description doc should be more accessible, eg. alt text of that column
- the Report documentation should also be labeled as such and easily accessible within markup
- Use headings only for real headings: could we use them better within Piwik software?
- general tip: let's not write "click here" and click proper labels to all links

**Summary**

The website piwik.org had fairly good standards of web content usability. The Piwik platform was also quite understandable and usable, despite a few improvements to make.

In general, if we write clean HTML markup (use Heading properly, use alt text, proper lists markup, etc.) the UI will be rather accessible by default. Also he mentioned that, **if the app can be used with the keyword only, then it is a great start!**. We also make great use of table markups (Table Heading th) and alt image text, which were big bonuses for accessibility.

We hope to keep learning about making Piwik more accessible and implement these improvements  above so that more people around the world including blind users and people with special needs, can enjoy the power of Piwik, the open analytics platform!

**Comment by mattab** [comment]
Hi guys, 
FYI we are working on Accessbility at the moment. We've created some issues linked here (you can see just above this comment).

**Comment by tassoman** [comment]
Got it, thank you!

**Comment by mattab** [comment]
FYI we had a 1 hour session with Julius, a user who is blind, and showed us the experience of using Piwik with screen reader, and it really helped us understand the difficulties in using Piwik! Let's make Piwik accessible.

Photo of session at: https://www.flickr.com/photos/4nitsirk/16251955566

**Comment by jdeyla** [comment]
Hi
I will face tickets in the next few days.
I’m very glad you had the chance to put your ears  in a blind guy screen reader,
it’s the only way to understand.


But think he’s just one user, with his expertise with technologies and web: just only one. 
Listen to him, but filter suggestions: you are the architect,




Jacopo Deyla

**Comment by mattab** [comment]
Hi everyone, we have made some progress on some of the most obvious and useful accessibility improvements. List of closed issues : 
![accessibility progress](https://cloud.githubusercontent.com/assets/466765/6077822/a4f81e08-ae57-11e4-97e8-e0e94e4ceda7.png)


(this was done with the help of high school students and Piwik team as part of the [Catalyst academy](https://piwik.org/blog/2015/01/catalyst-academy/))

we will release a beta in the next few days which will include those improvements. in the meantime you can test [Piwik from Git](http://piwik.org/faq/how-to-install/faq_18271/)  and let us know here any feedback!

**Comment by jdeyla** [comment]
Great job!






Jacopo Deyla

**Comment by jdeyla** [comment]
Hi @mattab I'd like to check #a11y issues on the latest beta release. Is there a public url and a demo user I can use in order to verify what's benn done? thx

**Comment by mattab** [comment]
Hi @jdeyla sure - you can test the latest version of Piwik on this demo: http://demo.piwik.org/

it would be great to hear from you and the results of your tests!

**Comment by jdeyla** [comment]
I thought it was the last stable,
not the last beta. 
Good and thank you!


>

**Comment by mattab** [comment]
Hi guys,

What do you think should be our next improvement regarding Accessibility?

Basically I'm asking anyone that need Piwik to be accessible to comment here with the most important next thing to fix. I'm sure there are plenty of annoyances of the product and it would be great to focus on the urgent ones :+1:

**Comment by tassoman** [comment]
Ciao @mattab , I'm aware of the work you're doing in the lateral navigation menu refactoring. Maybe another important step could be the calendar accessibility improvement and the select combo box used for switching websites.

Nowadays we're going to get rid of PHP 5.3 in our testing server so we can switch to a current Piwik installation. After this we could start pushing some code on a11y tasks. But this couldn't happen before issue's 7181 solution

**Comment by tassoman** [comment]
I'm reporting a strange behavior of the new sidebar menu: when you click a onepaged level1 (es: Dashboard, Goals) you get a page load. When you click a multipage level1 (es: Visitors, Actions) you only toggle its second level menu.

So now I'm unsure if the multipaged level1 links should have a "Actions widget" page, or if the onepaged level1 link "Goals" should toggle its menu only.

My choice would be the first case, so we should have routing for multipaged menus, routed to its first level2. For example I should land on "Overview" 2level page just by clicking "Actions"

**Comment by mattab** [comment]
> For example I should land on "Overview" 2level page just by clicking "Actions"

We changed this back on purpose, see discussion in https://github.com/piwik/piwik/issues/9007

**Comment by mattab** [comment]
It would be awesome to continue our work on Accessibility. If you want to help please take a look at all issues tagged with Accessibility here: https://github.com/piwik/piwik/labels/c%3A%20Accessibility
